### PR TITLE
RavenDB-8181 Auto complete RQL to fix the completions

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Raven.Studio/typescript/common/queryCompleter.ts [hunk 1]
```diff
@@ -468,24 +461,35 @@ class queryCompleter {
...
             }
-            return word;
+            return keyword;
         }))
     }
 
-    private isLetterOrDigit(str: string) {
-        // TODO: Add support for more letters in other lanuagse.
-        return /^[0-9a-zA-Z_@]+$/.test(str)
+    private escapeCollectionOrFieldName(name: string) : string {
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/queryCompleter.ts]: this can be static method, right?

File: src/Raven.Studio/typings/_studio/dto.d.ts [hunk 2]
```diff
@@ -317,12 +317,33 @@ interface layoutable {
...
 interface autoCompleteWordList {
-    caption: string; 
+    caption?: string; 
     value: string; 
     score: number; 
     meta: string 
 }
 
+interface autoCompleteLastKeyword {
+    keywordsBefore: string[],
```
Comment (ml054) [inline src/Raven.Studio/typings/_studio/dto.d.ts:327]: this struct becomes too complex imho...

File: src/Raven.Studio/typescript/common/queryCompleter.ts (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Studio/typescript/common/queryCompleter.ts]: I changed it to be static method.
But in general I do not like the static syntax of typescript so this is why I used instance method here by default.

File: src/Raven.Studio/typings/_studio/dto.d.ts (hunk 2 above)
Comment (fitzchak) [reply src/Raven.Studio/typings/_studio/dto.d.ts:327]: I would simplify it later on.

File: src/Raven.Studio/typescript/common/queryCompleter.ts (hunk 1 above)
Comment (ml054) [reply src/Raven.Studio/typescript/common/queryCompleter.ts]: I agree that syntax for static method in TS is weird. :(

**Pull request merged**